File: 724-find-pivot-index/724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int total = 0, left = 0, right = 0;
        for(int i = 0; i < nums.size(); i++)
            total += nums[i];
        for(int i = 0; i < nums.size(); i++){
            right = total - nums[i] - left;
            if(left == right)
                return i;
            left += nums[i];
        }
        return -1;
    }
};

File: 724-find-pivot-index/724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    var totalSum = 0, left = 0, right = 0;
    for(let i = 0; i < nums.length; i++)
        totalSum += nums[i];
    for(let i = 0; i < nums.length; i++){
        right = totalSum - nums[i] - left;
        if(left == right){
            return i;
        }
        left += nums[i];
    }
    return -1;
};